ironfish-cli/src/utils/chainport.ts

```diff
@@ -0,0 +1,71 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import axios from 'axios'
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+const SUPPORTED_NETWORKS = ['BSC', 'OPTIMISM', 'POLYGON', 'ARBITRUM', 'ETHEREUM']
+
+type BridgeNetworkMap = {
+  [key: string]: ChainportBridgeNetwork
+}
+
+export type ChainportBridgeNetwork = {
+  chainport_network_id: number
+  shortname: string
+  name: string
+  chain_id: number
+  explorer_url: string
+  label: string
+  blockchain_type: string
+  native_token_symbol: string
+  network_icon: string
+}
+
+/**
+ * Example response from the API:
+ * {
+    "label": "Ethereum",
+    "native_token_symbol": "ETH",
+    "button_colour": "grey",
+    "minimum_confirmation": 6,
+    "explorer_url": "https://sepolia.etherscan.io/",
+    "public_rpc": "https://nostalgic-agnesi:pushup-gamma-juice-common-drank-deploy@nd-533-089-070.p2pify.com/4b84120dbf2064427cfba156ded6d8f0",
+    "chainport_network_id": 2,
+    "shortname": "ETH",
+    "name": "ETHEREUM",
+    "utils_contract": "0xec3f107e26f9246ae1cfa98170b574a341481be1",
+    "chain_id": 11155111,
+    "blockchain_type": "evm",
+    "network_icon": "https://chainport-public-files-preprod.s3.amazonaws.com/3f07fedb-9d62-4756-8cd9-23f2c1733c9b.svg",
+    "bridges": {
+      "ChainportSideBridge": "0x5d3a13fb86f583ba3b45a1d8db88bdb0678121d2",
+      "ChainportMainBridge": "0x2a72d0f934896030241d44c4e12216ebb4229b4e",
+      "UsdcBridge": "0x0e5526f15f2d5db7520c71e32ab15a5e7ad58616",
+      "StablesWrapper": "0x365d5d02db9d1fa550750d784d4b542457406b80",
+      "StablesSwap": "0xd790849499b9eb73fb297115dadfdd2aef6a5782"
+    }
+  }
+ */
+
+export const getSupportedNetworkDetails = async () => {
+  const get = 'https://preprod-api.chainport.io/meta'
+  const response: {
+    data: {
+      cp_network_ids: BridgeNetworkMap
+    }
+  } = await axios.get(get)
+
+  const networks = response.data.cp_network_ids
+
+  const filteredNetworks = Object.keys(networks).reduce((acc, key: string) => {
+    if (SUPPORTED_NETWORKS.includes(networks[key].name)) {
+      acc[key] = networks[key]
```

Comment:
Shouldn't we warn/error here if we expect a supported network isn't found in the response?